src/Bicep.LangServer/Snippets/SnippetsProvider.cs

```diff
@@ -135,37 +140,53 @@ private void CacheResourceDeclarationAndDependencies(string template, string man
                 {
                     if (declaredSymbol.Type is TypeSymbol typeSymbol && typeSymbol.TypeKind != TypeKind.Error)
                     {
-                        string type = typeSymbol.Name;
-                        CacheResourceDeclaration(resourceDeclarationSyntax, type, template, description);
-                        CacheResourceDependencies(kvp.Value, template, type);
+                        CacheResourceDeclaration(resourceDeclarationSyntax, typeSymbol, template, description, manifestResourceName);
+                        CacheResourceDependencies(typeSymbol, kvp.Value, template);
                     }
                 }
             }
         }
 
-        private void CacheResourceDeclaration(ResourceDeclarationSyntax resourceDeclarationSyntax, string type, string template, string description)
+        private void CacheResourceDeclaration(ResourceDeclarationSyntax resourceDeclarationSyntax, TypeSymbol typeSymbol, string template, string description, string manifestResourceName)
         {
-            if (!resourceTypeToBodyMap.ContainsKey(type))
+            string type = typeSymbol.Name;
+
+            if (!resourceTypeInfoMap.ContainsKey(type))
             {
                 TextSpan bodySpan = resourceDeclarationSyntax.Value.Span;
                 string bodyText = template.Substring(bodySpan.Position, bodySpan.Length);
+                string prefix = Path.GetFileNameWithoutExtension(manifestResourceName);
+                TextSpan resourceDeclarationSyntaxNameSpan = resourceDeclarationSyntax.Name.Span;
+                string identifier = template.Substring(resourceDeclarationSyntaxNameSpan.Position, resourceDeclarationSyntaxNameSpan.Length);
 
-                resourceTypeToBodyMap.TryAdd(type, (bodyText, description));
+                resourceTypeInfoMap.TryAdd(type, (prefix, identifier, bodyText, description));
             }
         }
 
-        private void CacheResourceDependencies(ImmutableHashSet<ResourceDependency> resourceDependencies, string template, string resourceType)
+        private void CacheResourceDependencies(TypeSymbol childTypeSymbol, ImmutableHashSet<ResourceDependency> resourceDependencies, string template)
         {
             if (resourceDependencies.Any())
             {
                 StringBuilder sb = new StringBuilder();
+
                 foreach (ResourceDependency resourceDependency in resourceDependencies)
                 {
+                    string parentType = resourceDependency.Resource.Type.Name;
+
+                    if (resourceTypeToChildTypeSymbolsMap.ContainsKey(parentType))
+                    {
+                        resourceTypeToChildTypeSymbolsMap[parentType].Add(childTypeSymbol);
+                    }
+                    else
+                    {
+                        resourceTypeToChildTypeSymbolsMap.TryAdd(parentType, new List<TypeSymbol> { childTypeSymbol });
+                    }
```

Comment:
You're using a concurrent dictionary which implies you're concerned about thread-safety. However, I don't think the `.Add()` method on the list is technically thread-safe.

If you want to handle it in a thread-safe manner, then you can use `ImmutableArray<TypeSymbol>` for the dictionary value type, and replace this logic with `AddOrUpdate`:

```csharp
resourceTypeToChildTypeSymbolsMap.AddOrUpdate(
    parentType,
    _ => new() { childTypeSymbol },
    (_, children) => children.Add(childTypeSymbol));
```

Reply:
That makes sense. Updated here:
https://github.com/Azure/bicep/commit/0c0c664878dbd906644c181a547c42f854f4d5f3